src/java/fr/paris/lutece/plugins/forms/export/ExportServiceManager.java

```diff
@@ -170,4 +174,37 @@ else if ( config.getField( ).equals( ID_TECH_KEY ) )
         }
         return configList;
     }
+    
+    public List<String> generateNameComponents( Form form, FormResponse response, List<FormExportConfig> configList )
+    {
+        List<String> valueList = new ArrayList<>( );
+        
+        for ( FormExportConfig config : configList )
+        {
+            if ( config.getField( ).equals( FORM_NAME_KEY ) )
+            {
+                valueList.add( form.getTitle( ) );
+            }
+            else if ( config.getField( ).equals( ID_TECH_KEY ) )
+            {
+                valueList.add( String.valueOf( response.getId( ) ) );
+            }
+            else
+            {
+                for ( FormResponseStep step : response.getSteps( ) )
+                {
+                    FormQuestionResponse formQuestionResponse = step.getQuestions( ).stream( ).filter( fqr -> fqr.getQuestion( ).getId( ) == Integer.parseInt( config.getField( ) ) )
+                            .findFirst( ).orElse( null );
+                    if ( formQuestionResponse != null )
+                    {
+                        IEntryTypeService entryTypeService = EntryTypeServiceManager.getEntryTypeService( formQuestionResponse.getQuestion( ).getEntry( ) );
+                        valueList.add( entryTypeService.getResponseValueForExport( formQuestionResponse.getQuestion( ).getEntry( ), null, formQuestionResponse.getEntryResponse( ).get( 0 ), null ) );
+                        break;
```

Comment:
Do you name files by a concatenation of the responses ?
If so:
 * how do you handle filesystem file name character constraints
 * how do you handle filesystem file name length limits ?

Reply:
The first point was taken into account in the last commit. For the second point, the functionality only allows you to select form fields to add their value to an export file title. The user can therefore estimate the length of the title that he will obtain with his configuration.